frontends/web/src/routes/exchange/components/sell.tsx

```diff
@@ -0,0 +1,216 @@
+      onSelectExchange(exchange[0].exchangeName);
+    }
+  }, [hasOnlyOneSupportedExchange, allExchangeDeals, selectedRegion, onSelectExchange]);
+
+  useEffect(() => {
+    if (!versionInfo) {
+      return;
+    }
+
+    if (versionInfo.canUpgrade) {
+      setExchangeNotAvailableReason('version');
+      return;
+    }
+
+    const noExchangeAvailable = !showMoonpay && !showPocket;
+    const hasNoSupportedAccount = !supportedAccounts.some(a => a.code === accountCode);
+
+
+    if (hasNoSupportedAccount) {
+      setExchangeNotAvailableReason('coinCode');
+      return;
+    }
+
+    if (noExchangeAvailable) {
+      setExchangeNotAvailableReason('region');
+      return;
+    }
+
+    setExchangeNotAvailableReason(undefined);
+
+  }, [showMoonpay, showPocket, supportedAccounts, accountCode, versionInfo]);
+
+  const constructErrorMessage = () => {
+    switch (exchangeNotAvailableReason) {
+    case 'coinCode':
+      return t('No sell options available for this coin type.');
+    case 'region':
+      return t('No sell options available for this region.');
+    case 'version':
+      return t('Update firmware to sell.');
+    default:
+      return 'not-supported';
+    }
+  };
+
+  const showUpgradeFirmwareButton = versionInfo && versionInfo.canUpgrade;
+
+  if (!versionInfo) {
+    return null;
+  }
+
+  return (
+    <>
+      <div className={style.innerRadioButtonsContainer}>
+        {exchangeNotAvailableReason && (
+          <div className="flex flex-column">
+            <p className={style.noExchangeText}>{constructErrorMessage()}</p>
+            {showUpgradeFirmwareButton &&
+              <Button
+                className={style.updateButton}
+                onClick={() => {
+                  setFirmwareUpdateDialogOpen(true);
+                  navigate(`/settings/device-settings/${deviceIDs[0]}`);
+                }}
+                transparent>
+                {t('Update now')}
+              </Button>
+            }
+          </div>
+        )}
+        {!exchangeNotAvailableReason && allExchangeDeals && allExchangeDeals.exchanges.map(exchange => exchange.supported && (
+          <ExchangeSelectionRadio
+            key={exchange.exchangeName}
+            id={exchange.exchangeName}
+            exchangeName={exchange.exchangeName}
+            deals={exchange.deals}
+            checked={selectedExchange === exchange.exchangeName}
+            onChange={() => {
+              onSelectExchange(exchange.exchangeName);
+            }}
+            onClickInfoButton={() => {}}
+          />
+        ))}
+      </div>
+      {!exchangeNotAvailableReason &&
+          <div className={style.buttonsContainer}>
+            {supportedAccounts.length > 1 &&
+                    <Button
+                      className={style.buttonBack}
+                      secondary
+                      onClick={() => navigate('/buy/info')}>
+                      {t('button.back')}
+                    </Button>}
```

Comment:
nit: strange indentation, also for readability  `&& ( <JSXstuff> )` instead of `&& <JSXstuff>`